### DISCUSSION TITLE
v2.2.4 延迟测试，测试结果异常

### DISCUSSION THREAD

**Discussion by redcats2** [XIU2/CloudflareSpeedTest#459] [讨论]
数月前：结果正常，CF|CFT都能扫描到最好的在30-40ms的CDN server。

======================
最近两个月：无论是CF还是CFT，都无法正常获取延迟，扫出来的头部结果很多都是无法ping通，或是延迟200+

![2023-11-12_13-52-35](https://github.com/XIU2/CloudflareSpeedTest/assets/26648188/8b190899-7be9-44df-829f-4c13dc096657)

**Comment by redcats2** [comment]
Remark：已自建，上图为仅测试延迟

**Comment by charSLee013** [comment]
我发现个问题，**平均延迟低到**1～3ms之间，这基本上不太现实的

参考之前的[关于假墙的讨论](https://github.com/XIU2/CloudflareSpeedTest/issues/111)

要不路由器开了**clash TUN**之类的模式，要不就是碰到假墙了

**Reply by redcats2** [reply]
相同配置我在日本的机器测了，Top10都是3ms以下，不过都是真的，试了Top5，都是可用的。可以推断应该是墙的难问题。

不过关于假墙，我加了tl和tll，圈定范围之后，得到的是“符合选定范围的‘虚假结果’”。。。就比较无语

暂时的解决办法是用在日本的机器上测出的Top5 CDNIP，也还堪用

**Comment by redcats2** [comment]
圈定范围之后，得到的是“符合选定范围的‘虚假结果’”

![4](https://github.com/XIU2/CloudflareSpeedTest/assets/26648188/e7410667-e539-4b8e-ad51-6b5307cceba7)

**Reply by charSLee013** [reply]
所以是墙外VPS测速之后，拿测量好的IP到墙内用么？

这样的话有个问题，因为cloudflare CDN 是开启了 **anycast** 的，这意味着**不同地区得出的路由结果是完全不太一样的**

尤其是墙内**限制出海节点**且**墙中墙**的情况下，墙外的测试不代表墙内也能正常使用，而且墙外不同地区的测速结果也是不通用的

参考这个[issues](https://github.com/XIU2/CloudflareSpeedTest/issues/360)，相同IP地址会因为路由变化而路由到不同的地区的

**Reply by redcats2** [reply]
我用的是aws的cloudfront，昨天用日本VPS的top10，加上本地ICMPping确认，还行，油管能稳定在20-40w

**Reply by redcats2** [reply]
而且这个假墙太诡异了，和你说的这个https://github.com/XIU2/CloudflareSpeedTest/issues/111 还不完全一样，所以等等看，看是我个人问题还是地区普遍现象
